streamlabs/obs-studio-node
# Node obs log performance 2

### Description
This change disables LOG_DEBUG messages for log files created in `slobs-client\node-obs\logs` for the release build. It does not touch any other log destination like `OBS_API::LogReport` and `OutputDebugStringW`.
You can still enable LOG_DEBUG by creating a file _slobs-client/**enable-debug-logs**_ even if you have compiled the application with the release config. See the full typical paths below:
* **Windows**: _C:\Documents and Settings\USER_NAME\AppData\Roaming\slobs-client\\**enable-debug-logs**_
* **macOS**: _/Users/USER_NAME/Library/Application Support/slobs-client/**enable-debug-logs**_

### Motivation and Context
By default, we should not write debug logs to user machines. They do not need it.

### How Has This Been Tested?
1. Start Streamlabs Desktop, do some actions, close it.
2. View newly created logs in the folder:
* **Windows**: C:\Documents and Settings\YOUR_USERNAME\AppData\Roaming\slobs-client\node-obs\logs
* **macOS**: /Users/YOUR_USERNAME/Library/Application Support/slobs-client/node-obs/logs
3. Check that the file lines do not have `[Debug]`. They should have something like `[INFO]`, `[ERROR]`, etc. The old files may still have `[Debug]`.
4. Create the **enable-debug-logs** file in the appropriate folder (see examples above).
5. Start Streamlabs Desktop again, do some actions, close it.
6. Now you may see the `[Debug]` in the newly created log files.

### Types of changes
- New feature (it only affects developers)
- Performance enhancement (non-breaking change which improves efficiency)

### Checklist:
